tests/Integration/Behaviour/Features/Context/CustomerManagerFeatureContext.php

```diff
@@ -82,31 +84,46 @@ public function createACustomerUsingCommand($customerReference, TableNode $table
         $groupProvider = CommonFeatureContext::getContainer()->get('prestashop.adapter.group.provider.default_groups_provider');
         $defaultGroups = $groupProvider->getGroups();
 
+        // Check if all required fields were provided
         $mandatoryFields = [
             'firstName',
             'lastName',
             'email',
-            'password',
         ];
-
         foreach ($mandatoryFields as $mandatoryField) {
             if (!array_key_exists($mandatoryField, $data)) {
                 throw new Exception(sprintf('Mandatory property %s for customer has not been provided', $mandatoryField));
             }
         }
+        if (!array_key_exists('password', $data) && empty($data['isGuest'])) {
+            throw new Exception('Password must be provided, if creating a registered customer');
+        }
+
+        // Process data and apply fallbacks
+        $password = $data['password'];
+        $defaultGroupId = isset($data['defaultGroupId']) ? $data['defaultGroupId'] : $defaultGroups->getCustomersGroup()->getId();
```

Comment:
```suggestion
        $defaultGroupId = $data['defaultGroupId'] ?? $defaultGroups->getCustomersGroup()->getId();
```

nitpick, I know, but to give you an example of how you could do things a bit cleaner 👍🏻

Reply:
Honestly, I did not know how to use this syntax yet. I will keep it in mind 🚀